src/Microsoft.Dnx.Compilation.CSharp/RoslynProjectReference.cs

```diff
@@ -68,45 +69,95 @@ public IList<ISourceReference> GetSources()
                                      .ToList();
         }
 
-        public Assembly Load(IAssemblyLoadContext loadContext)
+        EmitResult EmitResourceAssembly(
+            AssemblyName assemblyName,
+            IEnumerable<ResourceDescriptor> resourceDescriptors,
+            AfterCompileContext afterCompileContext)
         {
-            using (var pdbStream = new MemoryStream())
-            using (var assemblyStream = new MemoryStream())
-            {
-                IList<ResourceDescription> resources = CompilationContext.Resources;
+            var resources = resourceDescriptors
+                .Select(res => new ResourceDescription(res.Name, res.StreamFactory, isPublic: true));
 
-                Logger.TraceInformation("[{0}]: Emitting assembly for {1}", GetType().Name, Name);
+            var mainCompilation = afterCompileContext.Compilation;
+            afterCompileContext.Compilation = CSharpCompilation.Create(
+                assemblyName.Name,
+                options: mainCompilation.Options);
 
-                var sw = Stopwatch.StartNew();
+            Logger.TraceInformation("[{0}]: Emitting assembly for {1}", GetType().Name, Name);
 
-                EmitResult emitResult = null;
+            var sw = Stopwatch.StartNew();
 
-                if (_supportsPdbGeneration.Value)
-                {
-                    emitResult = CompilationContext.Compilation.Emit(assemblyStream, pdbStream: pdbStream, manifestResources: resources);
-                }
-                else
-                {
-                    Logger.TraceWarning("PDB generation is not supported on this platform");
-                    emitResult = CompilationContext.Compilation.Emit(assemblyStream, manifestResources: resources);
-                }
+            var emitResult = afterCompileContext.Compilation.Emit(
+                    afterCompileContext.AssemblyStream,
+                    manifestResources: resources);
 
-                sw.Stop();
+            sw.Stop();
 
-                Logger.TraceInformation("[{0}]: Emitted {1} in {2}ms", GetType().Name, Name, sw.ElapsedMilliseconds);
+            Logger.TraceInformation("[{0}]: Emitted {1} in {2}ms", GetType().Name, Name, sw.ElapsedMilliseconds);
 
-                var diagnostics = CompilationContext.Diagnostics.Concat(
-                    emitResult.Diagnostics);
+            afterCompileContext.Diagnostics = CompilationContext.Diagnostics
+                .Concat(emitResult.Diagnostics)
+                .ToList();
 
+            return emitResult;
+        }
+
+        public Assembly Load(AssemblyName assemblyName, IAssemblyLoadContext loadContext)
```

Comment:
So I'm thinking that we shouldn't call into ICompileModules for resource assemblies. I know it just another assembly but it feels like ICompileModules would just start to break (like ours) if we just started to call it multiple times when we have resource assemblies. Can you remove that piece?